### ISSUE TITLE
TypeScript migration

### ISSUE THREAD

**Issue by ojeytonwilliams** [freeCodeCamp/freeCodeCamp#42256]
## Co-ordination

As discussed in the last contributors [meeting](https://forum.freecodecamp.org/t/office-hour-contributors-may-13-2021/458257/4) we're looking to migrate our client to TypeScript. 

In order to help people work in parallel this issue can be used to track who is working on what. I've put a checklist of all the unmigrated files at the bottom.  If you want to work on a file or some files, please edit in your github username next to the file(s) you want to work on.  If you don't have edit permission, please just reply to this issue @ me and I'll take care of it.

### Small is beautiful

If a file can be migrated on its own, that's perfect.  That makes the PR incredibly easy to review and minimises the chance we break anything.

### PropTypes can go

None of the components in the client dirs are intended to be consumed by JavaScript components - eventually everything will be `tsx`.  With that in mind static type checking should be enough.

### Ignore linting if necessary

Migration should be incremental and painless.  If you find that converting to TS requires a big rewrite, feel free to `@ts-ignore` or `eslint-disable` as needed.

It's more important that each change is easy to understand and review than it is it write perfect TS.  We'll get there eventually.

### More guidelines may get added

As we do this, chances are we'll revise the approach.  Any thoughts/comments/critiques are welcome, but I wanted to keep this straightforward for now.

<details> <summary> list of files left to migrate (collapsed because it's rather long) </summary>

- [ ] /client/utils/buildChallenges.js
- [ ] /client/utils/challengeTypes.js
- [ ] /client/utils/index.js
- [ ] /client/utils/stateManagement.js
- [ ] /client/utils/tags.js
- [ ] /client/utils/tags.test.js
- [ ] /client/utils/gatsby/challengePageCreator.js
- [ ] /client/utils/gatsby/index.js
- [ ] /client/utils/gatsby/layoutSelector.js
- [ ] /client/utils/gatsby/layoutSelector.test.js
- [ ] /client/src/html.js
- [ ] /client/src/utils/ajax.js
- [ ] /client/src/utils/algolia-locale-setup.js
- [ ] /client/src/utils/createTypes.js
- [ ] /client/src/utils/curriculum-helpers.test.js
- [ ] /client/src/utils/format.js
- [ ] /client/src/utils/format.test.js
- [ ] /client/src/utils/get-words.js
- [ ] /client/src/utils/handled-error.js
- [ ] /client/src/utils/handled-error.test.js
- [ ] /client/src/utils/index.js
- [ ] /client/src/utils/reallyWeirdErrorMessage.js
- [ ] /client/src/utils/report-error.js
- [ ] /client/src/utils/reportedErrorMessage.js
- [ ] /client/src/utils/scriptLoaders.js
- [ ] /client/src/utils/standardErrorMessage.js
- [ ] /client/src/utils/to-learn-path.js
- [ ] /client/src/utils/__fixtures/curriculum-helpers-css.js
- [ ] /client/src/utils/__fixtures/curriculum-helpers-html.js
- [ ] /client/src/utils/__fixtures/curriculum-helpers-javascript.js
- [ ] /client/src/utils/__fixtures/curriculum-helpers-remove-white-space.js
- [ ] /client/src/templates/Introduction/Intro.js
- [ ] /client/src/templates/Introduction/SuperBlockIntro.js
- [ ] /client/src/templates/Introduction/redux/index.js
- [ ] /client/src/templates/Introduction/components/Block.js
- [ ] /client/src/templates/Introduction/components/CertChallenge.js
- [ ] /client/src/templates/Introduction/components/CertificationCard.js
- [ ] /client/src/templates/Introduction/components/Challenges.js
- [ ] /client/src/templates/Introduction/components/ClaimCertSteps.js
- [ ] /client/src/templates/Introduction/components/SuperBlockIntro.js
- [ ] /client/src/templates/Challenges/video/Show.js
- [ ] /client/src/templates/Challenges/utils/build.js
- [ ] /client/src/templates/Challenges/utils/fetch-and-cache.js
- [ ] /client/src/templates/Challenges/utils/frame.js
- [ ] /client/src/templates/Challenges/utils/getTargetEditor.js
- [ ] /client/src/templates/Challenges/utils/index.js
- [ ] /client/src/templates/Challenges/utils/index.test.js
- [ ] /client/src/templates/Challenges/utils/postUpdate$.js
- [ ] /client/src/templates/Challenges/utils/worker-executor.js
- [ ] /client/src/templates/Challenges/utils/worker-executor.test.js
- [ ] /client/src/templates/Challenges/redux/code-lock-epic.js
- [ ] /client/src/templates/Challenges/redux/code-storage-epic.js
- [ ] /client/src/templates/Challenges/redux/completion-epic.js
- [ ] /client/src/templates/Challenges/redux/create-question-epic.js
- [ ] /client/src/templates/Challenges/redux/create-question-epic.test.js
- [ ] /client/src/templates/Challenges/redux/current-challenge-saga.js
- [ ] /client/src/templates/Challenges/redux/execute-challenge-saga.js
- [ ] /client/src/templates/Challenges/redux/index.js
- [ ] /client/src/templates/Challenges/rechallenge/builders.js
- [ ] /client/src/templates/Challenges/rechallenge/throwers.js
- [ ] /client/src/templates/Challenges/rechallenge/transformers.js
- [ ] /client/src/templates/Challenges/projects/SolutionForm.js
- [ ] /client/src/templates/Challenges/projects/Tool-Panel.js
- [ ] /client/src/templates/Challenges/projects/frontend/Show.js
- [ ] /client/src/templates/Challenges/projects/backend/Show.js
- [ ] /client/src/templates/Challenges/components/BreadCrumb.js
- [ ] /client/src/templates/Challenges/components/Challenge-Description.js
- [ ] /client/src/templates/Challenges/components/Challenge-Title.js
- [ ] /client/src/templates/Challenges/components/ChallengeTitle.test.js
- [ ] /client/src/templates/Challenges/components/CompletionModal.js
- [ ] /client/src/templates/Challenges/components/CompletionModal.test.js
- [ ] /client/src/templates/Challenges/components/CompletionModalBody.js
- [ ] /client/src/templates/Challenges/components/CompletionModalBody.test.js
- [ ] /client/src/templates/Challenges/components/HelpModal.js
- [ ] /client/src/templates/Challenges/components/Hotkeys.js
- [ ] /client/src/templates/Challenges/components/Output.js
- [ ] /client/src/templates/Challenges/components/Preview.js
- [ ] /client/src/templates/Challenges/components/PrismFormatted.js
- [ ] /client/src/templates/Challenges/components/ResetModal.js
- [ ] /client/src/templates/Challenges/components/Side-Panel.js
- [ ] /client/src/templates/Challenges/components/Test-Suite.js
- [ ] /client/src/templates/Challenges/components/Tool-Panel.js
- [ ] /client/src/templates/Challenges/components/VideoModal.js
- [ ] /client/src/templates/Challenges/classic/ActionRow.js
- [ ] /client/src/templates/Challenges/classic/DesktopLayout.js
- [ ] /client/src/templates/Challenges/classic/Editor.js
- [ ] /client/src/templates/Challenges/classic/EditorTabs.js
- [ ] /client/src/templates/Challenges/classic/MobileLayout.js
- [ ] /client/src/templates/Challenges/classic/MultifileEditor.js
- [ ] /client/src/templates/Challenges/classic/Show.js
- [ ] /client/src/resources/certAndProjectMap.js
- [ ] /client/src/resources/honesty-policy.js
- [ ] /client/src/redux/accept-terms-saga.js
- [ ] /client/src/redux/app-mount-saga.js
- [ ] /client/src/redux/cookieValues.js
- [ ] /client/src/redux/createStore.js
- [ ] /client/src/redux/donation-saga.js
- [ ] /client/src/redux/error-saga.js
- [ ] /client/src/redux/failed-updates-epic.js
- [ ] /client/src/redux/failed-updates-epic.test.js
- [ ] /client/src/redux/fetch-user-saga.js
- [ ] /client/src/redux/ga-saga.js
- [ ] /client/src/redux/ga-saga.test.js
- [ ] /client/src/redux/hard-go-to-epic.js
- [ ] /client/src/redux/index.js
- [ ] /client/src/redux/night-mode-saga.js
- [ ] /client/src/redux/propTypes.js
- [ ] /client/src/redux/report-user-saga.js
- [ ] /client/src/redux/rootEpic.js
- [ ] /client/src/redux/rootReducer.js
- [ ] /client/src/redux/rootSaga.js
- [ ] /client/src/redux/show-cert-saga.js
- [ ] /client/src/redux/update-complete-epic.js
- [ ] /client/src/redux/settings/danger-zone-saga.js
- [ ] /client/src/redux/settings/index.js
- [ ] /client/src/redux/settings/settings-sagas.js
- [ ] /client/src/redux/settings/update-email-saga.js
- [ ] /client/src/redux/settings/update-legacy-certificate-saga.js
- [ ] /client/src/pages/404.js
- [ ] /client/src/pages/certification.js
- [ ] /client/src/pages/challenges.js
- [ ] /client/src/pages/challenges.test.js
- [ ] /client/src/pages/donate.js
- [ ] /client/src/pages/email-sign-up.js
- [ ] /client/src/pages/index.js
- [ ] /client/src/pages/learn.js
- [ ] /client/src/pages/search.js
- [ ] /client/src/pages/settings.js
- [ ] /client/src/pages/unsubscribed.js
- [ ] /client/src/pages/update-email.js
- [ ] /client/src/pages/user.js
- [ ] /client/src/components/AppMountNotifier.js
- [ ] /client/src/components/RedirectHome.js
- [ ] /client/src/components/createExternalRedirects.js
- [ ] /client/src/components/createExternalRedirects.test.js
- [ ] /client/src/components/createLanguageRedirect.js
- [ ] /client/src/components/createLanguageRedirect.test.js
- [ ] /client/src/components/createRedirect.js
- [ ] /client/src/components/settings/About.js
- [ ] /client/src/components/settings/Certification.js
- [ ] /client/src/components/settings/Certification.test.js
- [ ] /client/src/components/settings/DangerZone.js
- [ ] /client/src/components/settings/DeleteModal.js
- [ ] /client/src/components/settings/Email.js
- [ ] /client/src/components/settings/Honesty.js
- [ ] /client/src/components/settings/Honesty.test.js
- [ ] /client/src/components/settings/Internet.js
- [ ] /client/src/components/settings/Portfolio.js
- [ ] /client/src/components/settings/Privacy.js
- [ ] /client/src/components/settings/ResetModal.js
- [ ] /client/src/components/settings/SectionHeader.js
- [ ] /client/src/components/settings/Theme.js
- [ ] /client/src/components/settings/ToggleSetting.js
- [ ] /client/src/components/settings/Username.js
- [ ] /client/src/components/search/WithInstantSearch.js
- [ ] /client/src/components/search/searchPage/EmptySearch.js
- [ ] /client/src/components/search/searchPage/NoResults.js
- [ ] /client/src/components/search/searchPage/SearchPageHits.js
- [ ] /client/src/components/search/searchBar/NoHitsSuggestion.js
- [ ] /client/src/components/search/searchBar/SearchBar.js
- [ ] /client/src/components/search/searchBar/SearchBar.test.js
- [ ] /client/src/components/search/searchBar/SearchHits.js
- [ ] /client/src/components/search/searchBar/SearchSuggestion.js
- [ ] /client/src/components/search/redux/index.js
- [ ] /client/src/components/profile/Profile.js
- [ ] /client/src/components/profile/Profile.test.js
- [ ] /client/src/components/profile/components/Camper.js
- [ ] /client/src/components/profile/components/Certifications.js
- [ ] /client/src/components/profile/components/HeatMap.js
- [ ] /client/src/components/profile/components/HeatMap.test.js
- [ ] /client/src/components/profile/components/Portfolio.js
- [ ] /client/src/components/profile/components/SocialIcons.js
- [ ] /client/src/components/profile/components/TimeLine.js
- [ ] /client/src/components/profile/components/TimeLine.test.js
- [ ] /client/src/components/profile/components/TimelinePagination.js
- [ ] /client/src/components/layouts/Certification.js
- [ ] /client/src/components/layouts/Default.js
- [ ] /client/src/components/layouts/Learn.js
- [ ] /client/src/components/layouts/index.js
- [ ] /client/src/components/landing/Landing.test.js
- [ ] /client/src/components/landing/index.js
- [ ] /client/src/components/landing/components/AsSeenIn.js
- [ ] /client/src/components/landing/components/BigCallToAction.js
- [ ] /client/src/components/landing/components/CampersImage.js
- [ ] /client/src/components/landing/components/Certifications.js
- [ ] /client/src/components/landing/components/LandingTop.js
- [ ] /client/src/components/landing/components/Testimonials.js
- [ ] /client/src/components/helpers/AvatarRenderer.js
- [ ] /client/src/components/helpers/ButtonSpacer.js
- [ ] /client/src/components/helpers/CurrentChallengeLink.js
- [ ] /client/src/components/helpers/FullWidthRow.js
- [ ] /client/src/components/helpers/ImageLoader.js
- [ ] /client/src/components/helpers/Link.js
- [ ] /client/src/components/helpers/Link.test.js
- [ ] /client/src/components/helpers/Loader.js
- [ ] /client/src/components/helpers/Loader.test.js
- [ ] /client/src/components/helpers/SkeletonSprite.js
- [ ] /client/src/components/helpers/SlimWidthRow.js
- [ ] /client/src/components/helpers/Spacer.js
- [ ] /client/src/components/helpers/ToggleButton.js
- [ ] /client/src/components/helpers/borderColorPicker.js
- [ ] /client/src/components/helpers/index.js
- [ ] /client/src/components/helpers/skeletonStyles.js
- [ ] /client/src/components/helpers/form/BlockSaveButton.js
- [ ] /client/src/components/helpers/form/BlockSaveWrapper.js
- [ ] /client/src/components/formHelpers/BlockSaveButton.js
- [ ] /client/src/components/formHelpers/BlockSaveButton.test.js
- [ ] /client/src/components/formHelpers/BlockSaveWrapper.js
- [ ] /client/src/components/formHelpers/BlockSaveWrapper.test.js
- [ ] /client/src/components/formHelpers/Form.js
- [ ] /client/src/components/formHelpers/Form.test.js
- [ ] /client/src/components/formHelpers/FormFields.js
- [ ] /client/src/components/formHelpers/FormValidators.js
- [ ] /client/src/components/formHelpers/index.js
- [ ] /client/src/components/SolutionViewer/ProjectModal.js
- [ ] /client/src/components/SolutionViewer/SolutionViewer.js
- [ ] /client/src/components/OfflineWarning/OfflineWarning.js
- [ ] /client/src/components/OfflineWarning/index.js
- [ ] /client/src/components/Map/Map.test.js
- [ ] /client/src/components/Map/index.js
- [ ] /client/src/components/Intro/Intro.test.js
- [ ] /client/src/components/Intro/index.js
- [ ] /client/src/components/Intro/components/IntroDescription.js
- [ ] /client/src/components/Header/Header.test.js
- [ ] /client/src/components/Header/index.js
- [ ] /client/src/components/Header/components/AuthOrProfile.js
- [ ] /client/src/components/Header/components/Login.js
- [ ] /client/src/components/Header/components/MenuButton.js
- [ ] /client/src/components/Header/components/NavLinks.js
- [ ] /client/src/components/Header/components/NavLogo.js
- [ ] /client/src/components/Header/components/UniversalNav.js
- [ ] /client/src/components/Header/components/UserState.js
- [ ] /client/src/components/FourOhFour/index.js
- [ ] /client/src/components/Footer/Footer.test.js
- [ ] /client/src/components/Footer/index.js
- [ ] /client/src/components/Flash/redux/index.js
- [ ] /client/src/components/Donation/DonateCompletion.js
- [ ] /client/src/components/Donation/DonateForm.js
- [ ] /client/src/components/Donation/DonationModal.js
- [ ] /client/src/components/Donation/DonationTextComponents.js
- [ ] /client/src/components/Donation/PayPalButtonScriptLoader.js
- [ ] /client/src/components/Donation/PaypalButton.js
- [ ] /client/src/components/Donation/assets/ApplePay.js
- [ ] /client/src/components/Donation/assets/GooglePay.js
- [ ] /client/src/client-only-routes/ShowCertification.js
- [ ] /client/src/client-only-routes/ShowProfileOrFourOhFour.js
- [ ] /client/src/client-only-routes/ShowProjectLinks.js
- [ ] /client/src/client-only-routes/ShowSettings.js
- [ ] /client/src/client-only-routes/ShowSettings.test.js
- [ ] /client/src/client-only-routes/ShowUnsubscribed.js
- [ ] /client/src/client-only-routes/ShowUser.js
- [ ] /client/src/client/frame-runner.js
- [ ] /client/src/client/workers/sass-compile.js
- [ ] /client/src/client/workers/test-evaluator.js
- [ ] /client/src/assets/images/components/AmazonLogo.js
- [ ] /client/src/assets/images/components/AppleLogo.js
- [ ] /client/src/assets/images/components/AsFeatureLogo.js
- [ ] /client/src/assets/images/components/GoogleLogo.js
- [ ] /client/src/assets/images/components/MicrosoftLogo.js
- [ ] /client/src/assets/images/components/SpotifyLogo.js
- [ ] /client/src/assets/images/components/index.js
- [ ] /client/src/assets/icons/APIIcon.js
- [ ] /client/src/assets/icons/Algorithm.js
- [ ] /client/src/assets/icons/Analytics.js
- [ ] /client/src/assets/icons/Caret.js
- [ ] /client/src/assets/icons/CertificationIcon.js
- [ ] /client/src/assets/icons/Clipboard.js
- [ ] /client/src/assets/icons/Cup.js
- [ ] /client/src/assets/icons/D3Icon.js
- [ ] /client/src/assets/icons/DefaultAvatar.js
- [ ] /client/src/assets/icons/DonateWithPayPal.js
- [ ] /client/src/assets/icons/Fail.js
- [ ] /client/src/assets/icons/FreeCodeCampLogo.js
- [ ] /client/src/assets/icons/GreenNotCompleted.js
- [ ] /client/src/assets/icons/GreenPass.js
- [ ] /client/src/assets/icons/Heart.js
- [ ] /client/src/assets/icons/Initial.js
- [ ] /client/src/assets/icons/IntroInformation.js
- [ ] /client/src/assets/icons/JavaScriptIcon.js
- [ ] /client/src/assets/icons/LinkButton.js
- [ ] /client/src/assets/icons/PythonIcon.js
- [ ] /client/src/assets/icons/ReactIcon.js
- [ ] /client/src/assets/icons/ResponsiveDesign.js
- [ ] /client/src/assets/icons/Shield.js
- [ ] /client/src/assets/icons/Spacer.js
- [ ] /client/src/assets/icons/TensorflowIcon.js
- [ ] /client/src/assets/icons/ToggleCheck.js
- [ ] /client/src/assets/icons/index.js
- [ ] /client/src/analytics/index.js
- [ ] /client/src/analytics/__mocks__/index.js
- [ ] /client/src/__tests__/integration/handled-error.test.js
- [ ] /client/src/__mocks__/challenge-nodes.js
- [ ] /client/src/__mocks__/fileMock.js
- [ ] /client/src/__mocks__/gatsby-link.js
- [ ] /client/src/__mocks__/gatsby.js
- [ ] /client/src/__mocks__/react-i18nextMock.js
- [ ] /client/src/__mocks__/styleMock.js
- [ ] /client/plugins/gatsby-remark-node-identity/gatsby-node.js
- [ ] /client/plugins/fcc-source-challenges/create-challenge-nodes.js
- [ ] /client/plugins/fcc-source-challenges/gatsby-node.js
- [ ] /client/i18n/config.js
- [ ] /client/i18n/configForTests.js
- [ ] /client/i18n/locales.test.js
- [ ] /client/i18n/schema-validation.js
- [ ] /client/i18n/validate-keys.js

</details>

**Comment by raisedadead** [comment]
Because this is a refactor we would want to rename the files to `kebab-case` where possible.

**Comment by Sembauke** [comment]
- camelCase
- PascalCase
- snake_case
- kebab-case

**Comment by krishna-saurav** [comment]
@ojeytonwilliams  I would like to contribute. Can I get edit permission for the above comment?

**Comment by ojeytonwilliams** [comment]
I'm not sure it's possible to make an issue publicly editable. @raisedadead do you know?

@krishna-saurav if you let us know which you're looking at, I can add your name to the checklist.

**Comment by krishna-saurav** [comment]
@ojeytonwilliams  Np !! I would like to take up below files :
/client/src/components/search/searchPage/EmptySearch.js
 /client/src/components/search/searchPage/NoResults.js
 /client/src/components/search/searchPage/SearchPageHits.js
 /client/src/components/search/searchBar/NoHitsSuggestion.js
 /client/src/components/search/searchBar/SearchBar.js
 /client/src/components/search/searchBar/SearchBar.test.js
 /client/src/components/search/searchBar/SearchHits.js
 /client/src/components/search/searchBar/SearchSuggestion.js
 /client/src/components/search/redux/index.js

**Comment by raisedadead** [comment]
Sorry, GitHub issues are editable by maintainers (our moderators) only. I think commenting on this thread should be fine and someone from the team would be able to ~~scratch off~~ the files from the OP once we have a PR merged.

**Comment by ojeytonwilliams** [comment]
I've edited the checklist for those files. Thanks for taking these on @krishna-saurav

**Comment by akshatgarg12** [comment]
I would like to work on these  @ojeytonwilliams 
 
/client/src/components/Header/index.js
/client/src/components/Header/components/AuthOrProfile.js
/client/src/components/Header/components/Login.js
/client/src/components/Header/components/MenuButton.js
/client/src/components/Header/components/NavLinks.js
/client/src/components/Header/components/NavLogo.js
/client/src/components/Header/components/UniversalNav.js
/client/src/components/Header/components/UserState.js

**Comment by ojeytonwilliams** [comment]
Awesome, thanks @akshatgarg12, I've edited you in.

**Comment by Dripcoding** [comment]
@ojeytonwilliams I'd like to work on these files: 

 /client/src/components/profile/Profile.js
 /client/src/components/profile/Profile.test.js
 /client/src/components/profile/components/Camper.js
 /client/src/components/profile/components/Certifications.js
 /client/src/components/profile/components/HeatMap.js
 /client/src/components/profile/components/HeatMap.test.js
 /client/src/components/profile/components/Portfolio.js
 /client/src/components/profile/components/SocialIcons.js
 /client/src/components/profile/components/TimeLine.js
 /client/src/components/profile/components/TimeLine.test.js
 /client/src/components/profile/components/TimelinePagination.js

**Comment by pedrocsilva** [comment]
Hey! Can I help migrate some files?

**Comment by Sembauke** [comment]
> Hey! Can I help migrate some files?

Sure go for it!

**Comment by ojeytonwilliams** [comment]
Thanks @Dripcoding, I've added your name to those files.

**Comment by pedrocsilva** [comment]
> > Hey! Can I help migrate some files?
> 
> Sure go for it! @pedrocsilva
> 
> Please specify which files you want to migrate to TypeScript.

I'll migrate these:

```
/client/src/components/landing/Landing.test.js
/client/src/components/landing/index.js
/client/src/components/landing/components/AsSeenIn.js
/client/src/components/landing/components/BigCallToAction.js
/client/src/components/landing/components/CampersImage.js
/client/src/components/landing/components/Certifications.js
/client/src/components/landing/components/LandingTop.js
/client/src/components/landing/components/Testimonials.js
```

**Comment by ojeytonwilliams** [comment]
Cheers, @pedrocsilva I've added your name, too.

**Comment by raisedadead** [comment]
This is going to be a merge-party issue. We will merge all linked PRs in a virtual party where we can have all contributors: just-chatting, reviewing PRs (resolving conflicts) and landing them.

**Comment by samarmohan** [comment]
I can do ` /client/utils/index.js`, `/client/utils/challengeTypes.js`, and `/client/utils/buildChallenges.js`

**Comment by awu43** [comment]
Hey, I'd like to take `client/src/pages`:

```
/client/src/pages/404.js
/client/src/pages/certification.js
/client/src/pages/challenges.js
/client/src/pages/challenges.test.js
/client/src/pages/donate.js
/client/src/pages/email-sign-up.js
/client/src/pages/index.js
/client/src/pages/learn.js
/client/src/pages/search.js
/client/src/pages/settings.js
/client/src/pages/unsubscribed.js
/client/src/pages/update-email.js
/client/src/pages/user.js
```

**Comment by ojeytonwilliams** [comment]
Thanks @awu43, I've edited you in :+1:

**Comment by Nrestrepo05** [comment]
Hey! I want to help with the following files:

/client/src/components/Donation/DonateCompletion.js
/client/src/components/Donation/DonateForm.js
/client/src/components/Donation/DonationModal.js
/client/src/components/Donation/DonationTextComponents.js
/client/src/components/Donation/PayPalButtonScriptLoader.js
/client/src/components/Donation/PaypalButton.js
/client/src/components/Donation/assets/ApplePay.js
/client/src/components/Donation/assets/GooglePay.js

**Comment by ojeytonwilliams** [comment]
Thanks @Nrestrepo05, I've edited you in too +1

**Comment by Nrestrepo05** [comment]
Sorry, I won't be able to work on this issue this week. Can I do it next week? If not, I'll look for a new issue later. Thanks!

**Comment by Nirajn2311** [comment]
Yea it's fine, there are a lot of files to be changed and everyone's just getting started with it.

**Comment by Murodjon000** [comment]
Hi @ojeytonwilliams may I help to migrate js to ts

**Comment by ShaunSHamilton** [comment]
Hello all,

The TypeScript Merge Party is scheduled to go down on the 25th of June, 2021, at 14:00 GMT.

Here is the forum post with more information: https://forum.freecodecamp.org/t/merge-party-typescript/465413

See you there 👀

**Comment by awu43** [comment]
7AM PT 😬. That's... less than ideal.

**Comment by Mvrs** [comment]
> Sorry, I won't be able to work on this issue this week. Can I do it next week? If not, I'll look for a new issue later. Thanks!

@ojeytonwilliams Hey, if you don't have time to do the Donation components can I give it a go?

**Comment by ojeytonwilliams** [comment]
I think, from the conversation in #contributors chat, that @Nrestrepo05 is still working on the Donation components.  Is that right @Nrestrepo05?  If so, it would be best to avoid duplicating each other's work.

If you're interested, @Mvrs , there are still a fair few unclaimed files - I'd be happy to assign you to any that you'd like to work on.

**Comment by ojeytonwilliams** [comment]
> 7AM PT . That's... less than ideal.

Sorry @awu43, this one isn't great for people in the states.  However, if we have it much later, then becomes too late for people in Asia.  If we do have a second party, we'll try and remember to make it a bit later so that different people can attend.

**Comment by Mvrs** [comment]
> I think, from the conversation in #contributors chat, that @Nrestrepo05 is still working on the Donation components.  Is that right @Nrestrepo05?  If so, it would be best to avoid duplicating each other's work.
> 
> 
> 
> If you're interested, @Mvrs , there are still a fair few unclaimed files - I'd be happy to assign you to any that you'd like to work on.

I can dig into the `client/src/helpers/**` methods.

**Comment by ojeytonwilliams** [comment]
> I can dig into the `client/src/helpers/**` methods.

Perfect, thanks.  I've made the edits.

**Comment by Nrestrepo05** [comment]
> > Sorry, I won't be able to work on this issue this week. Can I do it next week? If not, I'll look for a new issue later. Thanks!
> 
> @ojeytonwilliams Hey, if you don't have time to do the Donation components can I give it a go?

Hello, @Mvrs! Sorry, I'm using my free time to work on it 😕

**Comment by sharmaabhinav** [comment]
Hey! I want to help with the following files:

- [ ] /client/src/components/layouts/Default.js  
- [ ] /client/src/components/layouts/Learn.js
- [ ] /client/src/components/AppMountNotifier.js

**Comment by mnlightstone** [comment]
Hello! I'd love to help with this project. I'm new to the codebase, so can I just start off with these two files?

```
 /client/src/components/OfflineWarning/OfflineWarning.js
 /client/src/components/OfflineWarning/index.js
```

**Comment by ShaunSHamilton** [comment]
@sharmaabhinav I have added you to the list.

@mnlightstone I have added you to the list.

**Comment by samarmohan** [comment]
> 7AM PT 😬. That's... less than ideal.

Yeah I agree, for people on the east coast of the US that’s in the middle of the work day, is it possible we do the 26th (Saturday) instead?

**Comment by raisedadead** [comment]
Timezones are pretty hard. We would love to get everyone on the same meeting/call; the thing is, moving things is just going to be inconvenient for someone else.

We appreciate your understanding of this, and regardless of you being able to join, we want to let you know that your help and contribution to the code is much appreciated. 

Cheers.

**Comment by Mvrs** [comment]
@ojeytonwilliams was wondering do I make a commit after ever file change? (So the Git History looks clean)

**Comment by ShaunSHamilton** [comment]
@Mvrs For the most part, the PRs will be squashed. So, we are not much fussed about the branch history. What I did find, however, is Git is more likely to recognise a file as renamed, if the steps mentioned in the original post are followed. That is, it is always nice to preserve file history. So, it should be fine to  do all file renames in the same commit.

**Comment by Mvrs** [comment]
> @Mvrs For the most part, the PRs will be squashed. So, we are not much fussed about the branch history. What I did find, however, is Git is more likely to recognise a file as renamed, if the steps mentioned in the original post are followed. That is, it is always nice to preserve file history. So, it should be fine to  do all file renames in the same commit.

Thanks for clearing that up!

**Comment by Mvrs** [comment]
@ojeytonwilliams looked at the list and my assigned files mixed up with @thecodingaviator `/client/src/components/formHelpers/` and my `/client/src/components/helpers/**`. Just want to make sure my changes matter lol.

**Comment by RaymenDeol** [comment]
He - can I get started on the following please : 
` /client/src/components/layouts/Certification.js`

**Comment by thecodingaviator** [comment]
> @ojeytonwilliams looked at the list and my assigned files mixed up with @thecodingaviator `/client/src/components/formHelpers/` and my `/client/src/components/helpers/**`. Just want to make sure my changes matter lol.

Hey @Mvrs Please go through #42476 and see which files I've already worked on. Let me know which other ones you want to take up and I'll edit you in

**Comment by thecodingaviator** [comment]
> He - can I get started on the following please :
> ` /client/src/components/layouts/Certification.js`

@RaymenDeol I've edited you in. Go for it!

**Comment by Mvrs** [comment]
> > @ojeytonwilliams looked at the list and my assigned files mixed up with @thecodingaviator `/client/src/components/formHelpers/` and my `/client/src/components/helpers/**`. Just want to make sure my changes matter lol.
> 
> 
> 
> Hey @Mvrs Please go through #42476 and see which files I've already worked on. Let me know which other ones you want to take up and I'll edit you in

I believe you didn't touch the `helpers/` module yet. I'm almost done 🙏🏿

**Comment by thecodingaviator** [comment]
> > > @ojeytonwilliams looked at the list and my assigned files mixed up with @thecodingaviator `/client/src/components/formHelpers/` and my `/client/src/components/helpers/**`. Just want to make sure my changes matter lol.
> > 
> > 
> > Hey @Mvrs Please go through #42476 and see which files I've already worked on. Let me know which other ones you want to take up and I'll edit you in
> 
> I believe you didn't touch the `helpers/` module yet. I'm almost done 🙏🏿

I've worked on some files there like [this one](https://github.com/freeCodeCamp/freeCodeCamp/pull/42476/files#diff-181c3b3c416e2303f6190ba4e25b9984a143277ae2ddecc0fa8e93cbae9a5c07)

**Comment by moT01** [comment]
If anyone doesn't know about [this tool](https://tools.freecodecamp.org/) - it's kinda nice. You can type in a file or folder name/path and see if any open PR's touch that area of the code.

**Comment by Mvrs** [comment]
@thecodingaviator nice! I will avoid the form components then lol.

**Comment by Mvrs** [comment]
@ojeytonwilliams what else needs to be done?

**Comment by Mvrs** [comment]
> > > > @ojeytonwilliams looked at the list and my assigned files mixed up with @thecodingaviator `/client/src/components/formHelpers/` and my `/client/src/components/helpers/**`. Just want to make sure my changes matter lol.
> > > 
> > > 
> > > Hey @Mvrs Please go through #42476 and see which files I've already worked on. Let me know which other ones you want to take up and I'll edit you in
> > 
> > 
> > I believe you didn't touch the `helpers/` module yet. I'm almost done 🙏🏿
> 
> I've worked on some files there like [this one](https://github.com/freeCodeCamp/freeCodeCamp/pull/42476/files#diff-181c3b3c416e2303f6190ba4e25b9984a143277ae2ddecc0fa8e93cbae9a5c07)

@thecodingaviator Why did you continue to work on the files I already worked on? You shared a link of you doing two files in `/helpers/form/*` so I told I will not touch those. I was taking the rest lol. But, then you proceeded to do more helpers. C'mon my guy.

**Comment by thecodingaviator** [comment]
> > > > > @ojeytonwilliams looked at the list and my assigned files mixed up with @thecodingaviator `/client/src/components/formHelpers/` and my `/client/src/components/helpers/**`. Just want to make sure my changes matter lol.
> > > > 
> > > > 
> > > > Hey @Mvrs Please go through #42476 and see which files I've already worked on. Let me know which other ones you want to take up and I'll edit you in
> > > 
> > > 
> > > I believe you didn't touch the `helpers/` module yet. I'm almost done 🙏🏿
> > 
> > 
> > I've worked on some files there like [this one](https://github.com/freeCodeCamp/freeCodeCamp/pull/42476/files#diff-181c3b3c416e2303f6190ba4e25b9984a143277ae2ddecc0fa8e93cbae9a5c07)
> 
> @thecodingaviator Why did you continue to work on the files I already worked on? You shared a link of you doing two files in `/helpers/form/*` so I told I will not touch those. I was taking the rest lol. But, then you proceeded to do more helpers. C'mon my guy.

Hey, @Mvrs please go through the helpers I've edited/worked upon on [this link](https://github.com/freeCodeCamp/freeCodeCamp/pull/42476/files) and [this link](https://github.com/freeCodeCamp/freeCodeCamp/pull/42471/files). I am pretty sure I have only worked on those files that I have claimed in the file list above (this excludes any files I've edited to update imports but not migrated). However, if you feel like I've worked on a file that you claimed, I'd be happy to exclude it from my PRs. :)

**Comment by Mvrs** [comment]
@thecodingaviator Im saying two days ago you only had one link (which contained two files). I already informed you not to continue since I'm already doing the rest. The ones you were assigned was obviously a mistake in the list which was also brought up. Here are my [files](https://github.com/freeCodeCamp/freeCodeCamp/pull/42593). Just exclude the ones I've done that you've done.

**Comment by thecodingaviator** [comment]
> @thecodingaviator Im saying two days ago you only had one link (which contained two files). I already informed you not to continue since I'm already doing the rest. The ones you were assigned was obviously a mistake in the list which was also brought up. Here are my [files](https://github.com/freeCodeCamp/freeCodeCamp/pull/42593). Just exclude the ones I've done that you've done.

@Mvrs I'll remove those files from my PRs as a professional courtesy. However, I will strongly recommend using the tool Tom mentioned above to make sure you do not work on the same file(s) again if someone else already has an open PR. Happy coding :)

**Comment by Mvrs** [comment]
> > @thecodingaviator Im saying two days ago you only had one link (which contained two files). I already informed you not to continue since I'm already doing the rest. The ones you were assigned was obviously a mistake in the list which was also brought up. Here are my [files](https://github.com/freeCodeCamp/freeCodeCamp/pull/42593). Just exclude the ones I've done that you've done.
> 
> 
> 
> @Mvrs I'll remove those files from my PRs as a professional courtesy. However, I will strongly recommend using the tool Tom mentioned above to make sure you do not work on the same file(s) again if someone else already has an open PR. Happy coding :)

Note taken. Thanks for the tip and hacking coding as well 🙃

**Comment by ShaunSHamilton** [comment]
**Notes for Merge Party**
- [ ] First PR to be merged for `Spacer` changes: https://github.com/freeCodeCamp/freeCodeCamp/pull/42593
- [ ] Needs changes reverted to certain files: https://github.com/freeCodeCamp/freeCodeCamp/pull/42378
- [ ] Many files refer to `children` - majority of these should be `JSX.ElementChildrenAttribute` (most accurate typing, from what I have [read](https://www.typescriptlang.org/docs/handbook/jsx.html#children-type-checking)) _will be done post-party_
- [ ] Requires deliberation before merge: https://github.com/freeCodeCamp/freeCodeCamp/pull/42575
- [ ] Requires look over before merge: https://github.com/freeCodeCamp/freeCodeCamp/pull/42561
- [ ] Needs type package change reverted before merge: https://github.com/freeCodeCamp/freeCodeCamp/pull/42585
- [ ] Needs update before merge: https://github.com/freeCodeCamp/freeCodeCamp/pull/42287
- [ ] Thought needs to be given about the addition of the `react-instantsearch-core` library
- [ ] Needs update before merge: https://github.com/freeCodeCamp/freeCodeCamp/pull/42367
- [ ] Needs to not delete Italian work... : https://github.com/freeCodeCamp/freeCodeCamp/pull/42553

**Needs Rebase with `next`**
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42587
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42476
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42467
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42457
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42445
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42598
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42586
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42582
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42327

**Fully Ready-To-Go**
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42593
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42468
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42469
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42453
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42462
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/42587

**Comment by ShaunSHamilton** [comment]
Thank you, all those who contributed! Merge Party is over, but this issue still stands with many components still needing migrated.

Feel free to `@` us, if you want to work on a file/files 😃

**Comment by thecodingaviator** [comment]
![image](https://user-images.githubusercontent.com/34807532/123467306-fba8e180-d60d-11eb-9d4b-c934f25c3a8f.png)
139 files done!

**Comment by Mvrs** [comment]
@ShaunSHamilton are you going to make a list of the files that are left?

**Comment by ShaunSHamilton** [comment]
> @ShaunSHamilton are you going to make a list of the files that are left?

All the files that are unticked in the original post.

**Comment by thecodingaviator** [comment]
I think Shaun means unassigned rather than unticked^

**Comment by nhcarrigan** [comment]
Hey friends,

I'm going to try to land https://github.com/freeCodeCamp/freeCodeCamp/pull/42635 today, which will aim to resolve the existing linter issues, remove as many `eslint-disable` lines as possible, and get the client into a state where it builds.

Once this lands, we can rebase our open migration PRs and ensure we continue to pass our CI. 🙂

**Comment by thecodingaviator** [comment]
@nhcarrigan I'll have #42636 redone after that then?

**Comment by nhcarrigan** [comment]
Sounds good 🙂 

Mostly hoping to resolve the issue where your PR, for example, is failing checks because `next` is failing checks.

**Comment by awu43** [comment]
Round 2, here we go. I can do `/client/src/utils`
```
/client/src/utils/ajax.js
/client/src/utils/algolia-locale-setup.js
/client/src/utils/curriculum-helpers.test.js
/client/src/utils/format.js
/client/src/utils/format.test.js
/client/src/utils/get-words.js
/client/src/utils/handled-error.js
/client/src/utils/handled-error.test.js
/client/src/utils/index.js
/client/src/utils/reallyWeirdErrorMessage.js
/client/src/utils/report-error.js
/client/src/utils/reportedErrorMessage.js
/client/src/utils/scriptLoaders.js
/client/src/utils/standardErrorMessage.js
/client/src/utils/to-learn-path.js
```

It looks like `/client/src/utils/createTypes.js` has already been done (#42278).

**Comment by thecodingaviator** [comment]
@awu43 I’ve edited you in. Have a quick look the the OP to cross check but you should be good to go

**Comment by Mvrs** [comment]
@thecodingaviator are these open? `/client/src/templates/Challenges/redux/**`
I would like to take these.

**Comment by ShaunSHamilton** [comment]
@Mvrs No, I have taken all of the redux files.

You are welcome to look through the list in the original post to see which files do not have names next to them, and request your handle be added.

**Comment by Mvrs** [comment]
> @Mvrs No, I have taken all of the redux files.
> 
> 
> 
> You are welcome to look through the list in the original post to see which files do not have names next to them, and request your handle be added.

Got it! But they all have names next to them. How to tell if one is AFK or not? 🤣

**Comment by ShaunSHamilton** [comment]
> Got it! But they all have names next to them. How to tell if one is AFK or not? 🤣

@Mvrs I am not sure what you are looking at, because I can see a few dozen without names scattered around:
![image](https://user-images.githubusercontent.com/51722130/123616584-56a12b00-d7fe-11eb-8a66-a0ad8c49f2b7.png)

**Comment by Mvrs** [comment]
@ShaunSHamilton lol. I'll just take those

**Comment by thecodingaviator** [comment]
> @ShaunSHamilton lol. I'll just take those and these `/client/src/templates/Challenges/components/**`

@Mvrs I've assigned the following files to you:

![image](https://user-images.githubusercontent.com/34807532/123753764-d3ec9e80-d8d7-11eb-9c37-1abbc05942a0.png)

As for `/client/src/templates/Challenges/components/**`, not all of those files are available (those with usernames next to them are taken). Please let me know which specific ones you want

**Comment by raisedadead** [comment]
> ### 📢 Please target all further PRs to the `main` branch.

**Comment by awu43** [comment]
Ready for more files, I'll take `/client/utils/**`.

```
/client/utils/buildChallenges.js
/client/utils/challengeTypes.js
/client/utils/index.js
/client/utils/stateManagement.js
/client/utils/tags.js
/client/utils/tags.test.js
/client/utils/gatsby/challengePageCreator.js
/client/utils/gatsby/index.js
/client/utils/gatsby/layoutSelector.js
/client/utils/gatsby/layoutSelector.test.js
```

**Comment by Mvrs** [comment]
Just updating everyone!!! 

I'm still here and will have my files done soon 🚀

**Comment by vishwasnavadak** [comment]
Hi, Is it too late to say I will migrate below files? @ojeytonwilliams 
```
 /client/src/templates/Introduction/Intro.js
 /client/src/templates/Introduction/SuperBlockIntro.js
```
I don't see them assigned to anyone. So will be happy to help.

**Comment by ShaunSHamilton** [comment]
@vishwasnavadak I have edited your name against those files 👍

**Comment by swarup4741** [comment]
I would like to migrate `/client/src/components/landing/index.js` & `/client/src/components/landing/components/*`

**Comment by iigorcunha** [comment]
it's ok to get these files?

```plain text
client/src/redux/accept-terms-saga.js
client/src/redux/app-mount-saga.js
client/src/redux/cookieValues.js
client/src/redux/donation-saga.js

```

I did not see anyone get those files.

**Comment by Sembauke** [comment]
> it's ok to get these files?
@iigorcunha Sure go for it!

**Comment by iigorcunha** [comment]
Have any files still free to migrate?

**Comment by Sembauke** [comment]
> I can take up these files
> 
> ```
> /client/src/components/Map/Map.test.js
> /client/src/components/Map/index.js
> /client/src/components/Intro/Intro.test.js
> /client/src/components/Intro/index.js
> ```

I have added you in @sharmaabhinav!

**Comment by Sembauke** [comment]
> Have any files still free to migrate?

Hey @iigorcunha 
Here are some of the remaining tests
```
 /client/src/templates/Challenges/rechallenge/builders.js
 /client/src/templates/Challenges/rechallenge/throwers.js
 /client/src/templates/Challenges/rechallenge/transformers.js
 /client/src/templates/Challenges/components/HelpModal.js
 /client/src/templates/Challenges/components/Preview.js
 /client/src/templates/Challenges/components/Side-Panel.js
 /client/src/templates/Challenges/components/Test-Suite.js
 /client/src/templates/Challenges/components/Tool-Panel.js
 /client/src/templates/Challenges/components/VideoModal.js
```

**Comment by kwicherska** [comment]
Hi Guys! I would also like to help :) Could I take:
 /client/src/templates/Challenges/components/Test-Suite.js
 /client/src/templates/Challenges/components/Tool-Panel.js ?

**Comment by Sembauke** [comment]
> Hi Guys! I would also like to help :) Could I take:
> /client/src/templates/Challenges/components/Test-Suite.js
> /client/src/templates/Challenges/components/Tool-Panel.js ?

Sure I will ad you in!

**Comment by Sembauke** [comment]
Only a couple of files are not converted yet:
```
 /client/src/templates/Challenges/rechallenge/builders.js
 /client/src/templates/Challenges/rechallenge/throwers.js
 /client/src/templates/Challenges/rechallenge/transformers.js
 /client/src/components/search/WithInstantSearch.js
 /client/src/components/layouts/index.js
 /client/src/components/Intro/components/IntroDescription.js
 /client/src/components/Header/Header.test.js
 /client/src/client/frame-runner.js
 /client/src/client/workers/sass-compile.js
 /client/src/client/workers/test-evaluator.js
 ```
Feel free to take some of them!

**Comment by iigorcunha** [comment]
@Sembauke  as I saw in MathJax documentation, Versions greater than 3.0 are compatible with typescript. in this case the best  way to migrate this is upgrading MathJax? or just disable the lint in this lines? do you have any suggestion?

**Comment by iigorcunha** [comment]
@Sembauke can I discuss something about migrating files here? or have another way to do? it's better send a pull request and discuss there?

**Comment by Sembauke** [comment]
Opening a PR is the way to go so we can keep this issue clean 😁

**Comment by Ismailtlem** [comment]
Hi,
I am interested in taking those files : 
```  /client/src/components/search/WithInstantSearch.js
 /client/src/components/layouts/index.js
 /client/src/components/Intro/components/IntroDescription.js
 /client/src/components/Header/Header.test.js
```
Is there a time limit for pushing the converted files ?
-

**Comment by ojeytonwilliams** [comment]
@Ismailtlem no, no time limits at the moment.  Since it's a migration and not adding features or fixing critical bugs, there's no great time pressure.  I went ahead and added you to those files.

**Comment by kattmina** [comment]
> > Hi Guys! I would also like to help :) Could I take:
> > /client/src/templates/Challenges/components/Test-Suite.js
> > /client/src/templates/Challenges/components/Tool-Panel.js ?
> 
> Sure I will add you in!

Thank you, I will do it this weekend :)

**Comment by victor-duarte** [comment]
Hi @Sembauke, @ojeytonwilliams,

If still unassigned, I can pick up `/client/src/templates/Challenges/rechallenge/*` upon confirmation.

**Comment by Sembauke** [comment]
> Hi @Sembauke, @ojeytonwilliams,
> 
> If still unassigned, I can pick up `/client/src/templates/Challenges/rechallenge/*` upon confirmation.

Sure I will add you in!

**Comment by Sembauke** [comment]
Only 4 files left unassigned:
```
 /client/src/client/frame-runner.js
 /client/src/client/workers/sass-compile.js
 /client/src/client/workers/test-evaluator.js
 /client/src/components/search/WithInstantSearch.js
```

Feel free to assign yourself.

**Comment by awu43** [comment]
I'll take `/client/src/components/search/WithInstantSearch.js`.

**Comment by Sembauke** [comment]
> I'll take `/client/src/components/search/WithInstantSearch.js`.

I added you

**Comment by ojeytonwilliams** [comment]
Hey everyone, I just wanted to make the warning a bit more visible. Any file that's not *just* used by the client will not be migrateable (yet), due to the lack of tooling mentioned in https://github.com/freeCodeCamp/freeCodeCamp/issues/42892.  If a file is directly used by `gatsby-config`, `gatsby-node` or by anything outside of the `client` directory, then it will be used without compilation and so cannot be migrated yet.

Sorry about the inconvenience, but those files will need to wait for the additional tooling.

**Comment by victor-duarte** [comment]
@ojeytonwilliams, @Sembauke,

> Hey everyone, I just wanted to make the warning a bit more visible. Any file that's not _just_ used by the client will not be migrateable (yet), due to the lack of tooling mentioned in #42892. If a file is directly used by `gatsby-config`, `gatsby-node` or by anything outside of the `client` directory, then it will be used without compilation and so cannot be migrated yet.
> 
> Sorry about the inconvenience, but those files will need to wait for the additional tooling.

The files under ` /client/src/templates/Challenges/rechallenge/*` fall under the previous note.

```
 /client/src/templates/Challenges/rechallenge/builders.js
 /client/src/templates/Challenges/rechallenge/throwers.js
 /client/src/templates/Challenges/rechallenge/transformers.js
```

After migrating `builders.js` and running `npm run test` the following was thrown:
```
Error: Cannot find module '../rechallenge/builders'
Require stack:
- /Users/victorduarte/Documents/Projects/Forks/freeCodeCamp/client/src/templates/Challenges/utils/build.js
- /Users/victorduarte/Documents/Projects/Forks/freeCodeCamp/curriculum/test/test-challenges.js
```
The later (is outside client folder) ends up using from `builders.js`.

If any instructions are given on how to proceed will look into it. For the time being will stop migration of these.

**Comment by ManuSquall** [comment]
Hello world! Want to help too but don't figure out very well 😅; it's about updating js files to ts and make it work right?

**Comment by ManuSquall** [comment]
May i take ?
```
/client/src/client/frame-runner.js
```

**Comment by Sembauke** [comment]
Hey @victor-duarte,

The file that is is looking for `builder.js`  appears to be here:
https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/templates/Challenges/utils/build.js#L2

As Oliver said in the freecodecamp chat this file can not yet be migrated because of that.
After a solution is implemented for #42892 you may be able to migrate this file.

I am not sure about the other two files you mentioned maybe @ojeytonwilliams can give you an answer on that.

Sorry for the inconvenience,

As you see there  are still some files migratable:

```
 /client/src/client/workers/sass-compile.js
 /client/src/client/workers/test-evaluator.js
```
Feel free to take them.

**Comment by Sembauke** [comment]
> May i take ?
> 
> ```
> /client/src/client/frame-runner.js
> ```

Sure I will ad you in!

**Comment by kattmina** [comment]
> > > Hi Guys! I would also like to help :) Could I take:
> > > /client/src/templates/Challenges/components/Test-Suite.js
> > > /client/src/templates/Challenges/components/Tool-Panel.js ?
> > 
> > 
> > Sure I will add you in!
> 
> Thank you, I will do it this weekend :)

I have a delay - personal issues - I'm starting work on refactor this week, just letting you know. My apologies!

**Comment by ojeytonwilliams** [comment]
Thanks for keeping us posted, @kattmina.  Don't worry about the delay - take your time.

**Comment by meronokbay** [comment]
Can I pick up these two?
```
 /client/src/client/workers/sass-compile.js
 /client/src/client/workers/test-evaluator.js
```

**Comment by Sembauke** [comment]
> Can I pick up these two?
> 
> ```
>  /client/src/client/workers/sass-compile.js
>  /client/src/client/workers/test-evaluator.js
> ```

Yes I will ad you in!

**Comment by imrishabh18** [comment]
Are there any files left for migration?

**Comment by ojeytonwilliams** [comment]
> Are there any files left for migration?

Everything is currently assigned, @imrishabh18, but to keep making progress I suggest we wait until all the outstanding PRs are reviewed and merged.  Once that's done, enough time will have passed that we can remove the assignments and open them up again.

@Sembauke @ShaunSHamilton does that seem reasonable?

**Comment by mansijain980** [comment]
@ojeytonwilliams are there any files left unassigned and can be migrated to TS for JS?
Please let me know.

**Comment by ashiskumar-1999** [comment]
@ojeytonwilliams, Is there any component that needs to migrate from JS to TS or any component that has the priority to build we can discuss about it.

**Comment by ojeytonwilliams** [comment]
@mansijain980 @ashiskumar-1999 hey, thanks for the offer of help, but we're going to wait until the maintainers have had time to review and merge the existing PRs before we re-open assignments.

**Comment by ashiskumar-1999** [comment]
Hey @ojeytonwilliams, Thanks for the reply. 
Until is there any other issue that you wants to be solved currently?

**Comment by ojeytonwilliams** [comment]
Sure, we use "help wanted" and "first timers only" labels to highlight good issues for contributors to work on.  You can see them [here](https://github.com/freeCodeCamp/freeCodeCamp/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22+label%3A%22first+timers+only%22).  Also, the [contribution docs](https://contribute.freecodecamp.org/#/) have a lot of useful info about how to get started.

**Comment by Ismailtlem** [comment]
Hi,

Is typescript migration of those files assigned to anyone ? 
` /client/src/templates/Introduction/components/Block.js
 /client/src/templates/Introduction/components/CertChallenge.js
 /client/src/templates/Introduction/components/CertificationCard.js
 /client/src/templates/Introduction/components/Challenges.js
 /client/src/templates/Introduction/components/ClaimCertSteps.js
 /client/src/templates/Introduction/components/SuperBlockIntro.js
`
If not, i can take them

**Comment by vishwasnavadak** [comment]
Created the PR https://github.com/freeCodeCamp/freeCodeCamp/pull/43262 with the migrated files. 

The PR is marked as `has-translation`. I haven't made changes related to i18n, but I did pulled the latest changes from upstream with rebase. Could anyone help me understand what is the issue with the PR?

**Comment by ahmadabdolsaheb** [comment]
@vishwasnavadak, thanks for letting us know. Please see the comment on your PR.

**Comment by ojeytonwilliams** [comment]
@Ismailtlem nope, they're all yours. I've edited you in.  Thanks!

**Comment by nicolealdurien** [comment]
Hi! I just added a PR [(#43490)](https://github.com/freeCodeCamp/freeCodeCamp/pull/43490) migrating ActionRow.js. My first open source PR ever, so please let me know if there's anything I can improve!

**Comment by shubhangi013** [comment]
Hi, is  `/client/src/templates/Introduction/components/Challenges.js` assigned to anyone?
If not, I'd like to work on it. Thanks

**Comment by meronokbay** [comment]
Can I pick up these files?
```
 /client/src/templates/Challenges/classic/DesktopLayout.js
 /client/src/templates/Challenges/classic/EditorTabs.js
 /client/src/templates/Challenges/classic/MobileLayout.js
 /client/src/templates/Challenges/classic/MultifileEditor.js
```

**Comment by moT01** [comment]
I put you down for that file @shubhangi013
There's two open PR's for refactoring the DesktopLayout @meronokbay, we just forgot to mark it on here. I put you down for the other three

**Comment by Ismailtlem** [comment]
I can take those as well : 
```  /client/src/templates/Introduction/components/ClaimCertSteps.js
 /client/src/templates/Introduction/components/SuperBlockIntro.js
 /client/src/templates/Introduction/components/Block.js
```
If still not assigned to anyone

**Comment by ojeytonwilliams** [comment]
Sure @Ismailtlem, thanks for taking them on.

**Comment by ashiskumar-1999** [comment]
Hey @ojeytonwilliams, Can I take these for the typescript migration?
/client/src/components/landing/

**Comment by Sembauke** [comment]
Hey @ashiskumar-1999, 

These seem to be taken already, sorry!

**Comment by ashiskumar-1999** [comment]
@ojeytonwilliams 
Okay, you can mention the file names that needs the migration and not taken away by anyone.

**Comment by Sembauke** [comment]
@ashiskumar-1999  sure here are some:
```
 /client/src/components/SolutionViewer/ProjectModal.js
 /client/src/components/SolutionViewer/SolutionViewer.js
```

**Comment by ashiskumar-1999** [comment]
Thanks @Sembauke , I am taking it.

**Comment by ashiskumar-1999** [comment]
Hey @Sembauke, I have migrated the .js file to the .tsx file. I just want you to cross-check the code to find any ambiguities. Can I share the code sample here?

**Comment by thecodingaviator** [comment]
Hello @ashiskumar-1999 i believe you can create a PR, we’ll review it for ambiguities and request changes.

**Comment by RobertoMSousa** [comment]
Hello.
What are the rules to select the files that are not migrate yet. I started working from top of the components and some of them it seems that are already taken.

**Comment by ShaunSHamilton** [comment]
@RobertoMSousa Hello there. It does not appear there are any files available to migrate at the moment. That is, all files have been assigned to those who previously requested them.
